worlds/jakanddaxter/Regions.py

```diff
@@ -0,0 +1,210 @@
+        multiworld.completion_condition[player] = lambda state: state.can_reach(fb, "Region", player)
+
+    elif options.jak_completion_condition == CompletionCondition.option_open_100_cell_door:
+        multiworld.completion_condition[player] = lambda state: state.can_reach(fd, "Region", player)
+
+    # As a final sanity check on these options, verify that we have enough locations to allow us to cross
+    # the connector levels. E.g. if you set Fire Canyon count to 99, we may not have 99 Locations in hub 1.
+    verify_connector_level_accessibility(multiworld, options, player)
+
+    # Also verify that we didn't overload the trade amounts with more orbs than exist in the world.
+    verify_orbs_for_trades(multiworld, options, player)
+
+
+def verify_connector_level_accessibility(multiworld: MultiWorld, options: JakAndDaxterOptions, player: int):
+
+    # Set up a state where we only have the items we need to progress, exactly when we need them, as well as
+    # any items we would have/get from our other options. The only variable we're actually testing here is the
+    # number of power cells we need.
+    state = CollectionState(multiworld)
+    if options.enable_move_randomizer == EnableMoveRandomizer.option_false:
+        for move in move_item_table:
+            state.collect(JakAndDaxterItem(move_item_table[move], ItemClassification.progression, move, player))
+
+    thresholds = {
+        0: {
+            "option": options.fire_canyon_cell_count,
+            "required_items": {},
+        },
+        1: {
+            "option": options.mountain_pass_cell_count,
+            "required_items": {
+                33: "Warrior's Pontoons",
+                10945: "Double Jump",
+            },
+        },
+        2: {
+            "option": options.lava_tube_cell_count,
+            "required_items": {},
+        },
+    }
+
+    loc = 0
+    for k in thresholds:
+        option = thresholds[k]["option"]
+        required_items = thresholds[k]["required_items"]
+
+        # Given our current state (starting with 0 Power Cells), determine if there are enough
+        # Locations to fill with the number of Power Cells needed for the next threshold.
+        locations_available = multiworld.get_reachable_locations(state, player)
+        if len(locations_available) < option.value:
+            raise OptionError(f"Settings conflict with {option.display_name}: "
+                              f"not enough potential locations ({len(locations_available)}) "
+                              f"for the required number of power cells ({option.value}).")
+
+        # Once we've determined we can pass the current threshold, add what we need to reach the next one.
+        for _ in range(option.value):
+            state.collect(JakAndDaxterItem("Power Cell", ItemClassification.progression, loc, player))
+            loc += 1
+
+        for item in required_items:
```

Comment:
This is ultimately a test to make sure your power cell option does not exceed the number of locations available to put them. This is particularly important for Move Rando + No Orbsanity in Hub 1, where if you have no movement options unlocked, there are only 30 available locations, so you can easily accidentally overshoot this number. We still want to allow you to require more power cells of yourself, we just need to catch this scenario because it will cause seed generation to fail.

So you can think of each of the `thresholds` as a wall, and `required_items` are what you need to get past it. At the end of each `for k in thresholds` iteration, if you didn't `raise OptionError`, we know you got past that wall, so we "pretend" to add those items to your inventory in preparation for you to attempt to cross the next wall.

Reply:
I think `for _ in range(option.value):` might be a bug, because it would be adding too many power cells to your fake inventory. If you need to collect 20 cells in hub 1, and 45 in hub 2, this code would give you 65 cells heading into hub 3, but you should only have 45.